scripts/db-heal-test.py

```diff
@@ -0,0 +1,231 @@
+                process.terminate()
+                return checkpoint
+            if f"block {checkpoint + checkpoint_granularity}" in line:
+                # First checkpoint was found, we should start timer as it means block processing is running.
+                if start == 0.0:
+                    start = time.time()
+                checkpoint = checkpoint + checkpoint_granularity
+                print(f"Found new checkpoint {checkpoint}.")
+            # If process ends with error (return code 1) or either 'fail' or 'exit status' occurs in line exit script
+            if process.poll() == 1 or any(s in line for s in ["exit status", "fail"]):
+                print("Error occurred - printing output.log:")
+                with open(aida_log_file, 'r') as l:
+                    text = l.read()
+                    print(text)
+                os.chdir(current_dir)
+                return -1
+
+
+
+# First iteration command
+cmd = [
+    './build/aida-vm-sdb', 'substate', '--validate',
+    '--db-tmp', tmp_path, '--carmen-schema', '5', '--db-impl', 'carmen',
+    '--aida-db', aida_db_path, '--no-heartbeat-logging', '--track-progress',
+    '--archive', '--archive-variant', 's5', '--archive-query-rate', '200',
+    '--carmen-checkpoint-interval', str(checkpoint_granularity_flag), '--tracker-granularity',
+    str(checkpoint_granularity_flag), str(first_block), str(last_block)
+]
+
+os.chdir(aida_path)
+with open(aida_log_file, 'w') as f:
+    process = subprocess.Popen(cmd, stdout=f, stderr=subprocess.STDOUT)
+os.chdir(current_dir)
+
+print("Creating database with aida-vm-sdb...")
+
+# Start monitoring the log file
+latest_checkpoint = terminate_process_after(window, latest_checkpoint)
+
+# Wait for the first command to complete
+process.wait()
+
+if latest_checkpoint == -1:
+    sys.exit(1)
+
+# Find working directory
+working_dir = max(Path(tmp_path).iterdir(), key=os.path.getmtime)
+archive = working_dir / 'archive'
+live = working_dir / 'live'
+
+print("Testing db created, starting loop.")
+
+for i in range(1, number_of_iterations + 1):
+    last_working_dir = working_dir
+
+    # Find working dir - Aida copies db-src
+    working_dir = max(Path(tmp_path).iterdir(), key=os.path.getmtime)
+    archive = working_dir / 'archive'
+    live = working_dir / 'live'
+
+    # Dumb carmen's logs into a file to avoid spamming
+    c = open(carmen_log_file, 'w+')
+
+    # Restore Archive
+    result = subprocess.run(
+        ['go', 'run', './database/mpt/tool', 'reset', '--force-unlock', str(archive), str(latest_checkpoint)],
+        stdout=c,
+        stderr=c,
+        cwd=carmen_root)
+    if result.returncode != 0:
+        # Print carmen logs if interation fails
+        text = c.read()
+        print(text)
+        sys.exit(1)
+
+    # Export genesis to restore LiveDB
+    genesis = Path(tmp_path) / 'test_genesis.dat'
+
+    print(f"Restoration complete. Exporting LiveDB genesis block {latest_checkpoint}.")
```

Comment:
I think it is good to have experiment that continues sync after db was healed, but I think it would be worth having a separate script (pipeline?) for this. 

When I want to run experiment that creates a many CPs and kills the app often, verifying the reset works.  I think continuing with export/import/sync adds too much extra time

Reply:
Without the export/import/sync you actually dont test the whole `heal` feature. I think it should stay in here. Thats actually what we discussed with @HerbertJordan when I got assigned this task.

Reply:
I said split it in two scripts/pipelines as they can run at different pace, when it is split, we loose nothing

Reply:
@kjezek what would the first script do instead of continuing with the export/import/sync?

Reply:
I thought it could only do the db reset, possibly db reset +  export genesis. 
The idea was to speed-up this test, which stress-test checkpoint creation.

Reply:
In this case we would have to always start from an empty archive in each iteration?